### ISSUE TITLE
@Prop() schemaType on Map not working

### ISSUE THREAD

**Issue by kasium** [nestjs/mongoose#1931]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

Issue https://github.com/nestjs/mongoose/issues/413 is not yet fixed, even if it's marked as closed.
Because nobody reacted on the issue, that it is not solved, I opened this issue to notify you

### Minimum reproduction code

n/a

### Steps to reproduce

See https://github.com/nestjs/mongoose/issues/413

### Expected behavior

Return a map type instead of an object

### Package version

9.0.0

### mongoose version

6.6.5

### NestJS version

9.0.0

### Node.js version

14.21.2

### In which operating systems have you tested?

- [ ] macOS
- [X] Windows
- [ ] Linux

**Comment by kamilmysliwiec** [comment]
Please provide a [minimum reproduction repository](https://minimum-reproduction.wtf/) (Git repository/[StackBlitz](https://stackblitz.com/fork/github/nestjs/typescript-starter)/[CodeSandbox](https://codesandbox.io/s/github/nestjs/typescript-starter/tree/master) project).

**Comment by kasium** [comment]
I see, you cannot use `raw`, which makes sense

**Comment by nik2208** [comment]
any news about this topic?

I tried to make the same question in discord with no answer.
this is the question:

to store a ` { [key: string]: string }` object following the docs I've set my graphql field as 
```typescript
@Field(() => GraphQLJSON)
name: JSON;
```
(using JSON scalar) and on the schema the Mongoose Object is defined like
```typescript
@Prop({ type: SchemaTypes.Mixed })
```
this is the only working configuration I've come up with, but this doesn't involve any type of schema validation.
I expected I could do something like `@Prop({type:Map, of: String})`or `@Prop(raw({type:Map, of: String}))` but I get casting error:
```bash
validation failed: name: Cast to Map failed for value "[Object: null prototype] (type object) at path because of "TypeError" nestjs mongoose
```

is this expected?
